### DISCUSSION TITLE
Who are with us?

### DISCUSSION THREAD

**Discussion by syleeeee** [line/armeria#4626] [Show and tell]
# Are you using Armeria? 🙌 
Armeria has grown with the community by getting feedback. Promoting an open source based on the adoptors reference is powerful in every case. Support Armeria with your use case, and we'd love to show off all the great things of our contributors.   
Please add a comment below. 

## It would be great if you share ... 👀 
- Company name(+logo)
- Product/service name that adopted Armeria
- Any websites
- The initiative adopting Armeria
- Top 3 benefits in the use case

## Tada 🎉  
We are happy to announce that we now have the limited Armeria Adopters package! 
Tell us your use case of Armeria, and get the package for your team! (up to 3 packages per case)

![LINE_Armeria_6976](https://user-images.githubusercontent.com/6274210/211991204-6597efed-e4d7-45e0-afd7-57c5fb918a19.JPG)

_After leaving your use case, please submit **[this form](https://feedback.line.me/enquete/public/19644-O38pbLjG)** for the recipient information of the delivery._

**Comment by TheWeaVer** [comment]
We've got many supports from Armeria 🙌. Here is the our use case !


![LINE](https://user-images.githubusercontent.com/6274210/211973998-f88c3aa7-b437-4ec9-a092-dd5b460353f9.png)
* Company Name: LINE
* Product/service name that adopted Armeria: The Gateway using channel authn/z for LINE APIs
* Any websites: [LINE Developers](https://developers.line.biz/en/docs/)
* The initiative adopting Armeria
  * Reliable http/2 protocol support
  * Event loop based non-blocking server
  * Support migration from the [Spring Tomcat](https://github.com/line/armeria/tree/master/tomcat9)
* Top 3 benefits in the use case
  * Build client factory by decorators, protocol, discovery support
  * Easy use of the http implementation by wrapping http reactive interface 
  * Technical support at the slack channel: https://armeria.dev/s/slack

**Comment by ta7uw** [comment]
We've got many supports from Armeria 🙌. Here is the our use case !

- Company name(+logo): LINE
- Product/service name that adopted Armeria: LINEバイト (This service is provided in Japan only.)
- Any websites: https://baito.line.me/
- The initiative adopting Armeria
  - Active development and kind support
  - Spring Integration
  - De facto standard library for gRPC server/client in my company
- Top 3 benefits in the use case
  - Easily apply Armeria useful features to gRPC server/client
  - Informative documents and user-friendly Java API design
  - Easily integrate with Spring, including actuator and yaml-based configuration

**Comment by policeman-kh** [comment]
We've got many supports from Armeria 🙌. Here is the our use case !
* Company name(+logo): LINE
* Product/service name that adopted Armeria: LINE STORE 
* Any websites: https://store.line.me/
* The initiative adopting Armeria
  * Spring MVC Integration.
  * API client for building microservice, which implements circuit breaker and client-side load balancing.
  * Collecting metrics for both server and client.
* Top 3 benefits in the use case
  * Gentle and speedy technical support.
  * Good documentation and sample codes.
  * Enhance server and client by decorator more easily.

**Comment by tobias-** [comment]
* Company name(+logo) : YouCruit https://www.youcruit.com/images/youcruit-logo.svg
* Product/service name that adopted Armeria: YouCruit
* Any websites: https://www.youcruit.com
* The initiative adopting Armeria
  * Support grpc & normal web traffic on single port
  * Minimal dependencies to manage
* Top 3 benefits in the use case
  * Good Kotlin integration with coroutines
  * No auto-loading "magic" (we moved from Jersey)
  * The always helpful people in the slack channel

Other things that have been added during our journey with Armeria and have been helpful is
* GraphQL support
* OpenTelemetry
* Virtual office hours
.. and lots of things that I now take for granted or at least can't remember now

**Comment by chungonn** [comment]
- Company Name: Greenfossil Pte Ltd
-![logo](https://avatars.githubusercontent.com/u/115706110?s=96&v=4)

- Product/service name that adopted Armeria: 
	- Thorium, a Scala 3 microservice framework leveraging on Armeria. https://thoriumframework.dev
	- All our company's web apps, our company's website https://www.greenfossil.com, Thorium's website are developed using Thorium. 
	- The initiative adopting Armeria:
		- Support team is nothing short of amazing! They are incredibly helpful, their responses are detailed and prompt. 
		- They keep adding new features to better meet the needs of the Armeria's users.
		- Embrace Reactive Manifesto principles, rigorous and consistent application of Design Patterns
		- Architecture and APIs are very well-designed, it allows Thorium to dovetail easily to Armeria
	- Top 3 benefits in the use case:
		- Scala friendly
		- An enterprise grade swiss army knife network library which supports HTTP, gRPC, Thrift. 
		- Clean, detailed documentation makes onboarding easy

**Comment by mscheong01** [comment]
- Company name: Mathpresso
![image](https://user-images.githubusercontent.com/54794500/215680777-1250dacb-a775-4474-b934-060cd67fb8b8.png)
- Product/service name that adopted Armeria
    - Our company's SchoolExam service, that sells educational content to students on the QANDA app
- Any websites
    - https://qanda.ai/ko
- The initiative adopting Armeria
    - native gRPC HttpJsonTranscoding support - we're trying to follow the Google AIPs, and the transcoding feature is doing everything that is needed for us 😆 
    - spring webflux integration
    - both server/client side support for multiple protocols
- Top 3 benefits in the use case
    - easy to use decorator implementation
    - docservice that could serve both as a api documentation and a test client
    - speedy technical support on both the slack channel and github

**Comment by Dogacel** [comment]
![Logo - Color@3x](https://user-images.githubusercontent.com/7023385/215841920-8dba0843-2f36-478d-a958-320bfef2a60a.png)

* Company Name: Carbon Health
* Product/service name that adopted Armeria: 20+ microservices
  * Internal frameworks such as core libraries and identity service. 
  * Product microservices that manage prescriptions, appointments, bills and many other stuff.
* Any websites: https://carbonhealth.com/
* The initiative adopting Armeria
  * Serve both HTTP and gRPC.
  * Kotlin integration and coroutine support.
* Top 3 benefits in the use case
  * Decorators that help metric, error and trace collection.
  * Being able to HTTP and gRPC from the same service/port.
  * Open for contribution, easy to use and has good documentation.

And also thanks to the team for the fast responses 🙂 .

**Comment by kezhenxu94** [comment]
<img src="http://skywalking.apache.org/assets/logo.svg" alt="Sky Walking logo" height="90px" align="right" />

* Company Name: [many companies that use SkyWalking](https://skywalking.apache.org/users/)
* Product/service name that adopted Armeria: Apache SkyWalking
   * ElasticSearch client library based on their RESTful API.
   * GraphQL server to provide query APIs for UI.
   * The web app server hosting frontend static files and proxy the GraphQL requests. 
* Any websites: https://skywalking.apache.org/
* The initiative adopting Armeria
  
  * minimize dependencies.
  * Support both client and server with one dependency. 
* Top 3 benefits in the use case 
  * minimal dependencies to manage.
  * Being able to use HTTP, https and gRPC from the same service/port.
  * Open for contribution, awesome API design, easy to use and has good documentation.

**Comment by blue-hope** [comment]
* Company name: Mathpresso
* Logo: 
![image](https://user-images.githubusercontent.com/38323921/216988516-77ac214c-26e6-453f-9d27-8680f894c3a6.png)

* Product/service name that adopted Armeria
  * Our company's AD platform, which manage all ad assets(inhouse ads, sales ads, etc..) and supply them to our QANDA app.
* Any websites: [qanda.ai/ko](https://qanda.ai/ko)
* The initiative adopting Armeria
  * Spring webflux integration
  * client/server side support for http / gRPC
* Top 3 benefits in the use case
  * Both gRPC(server to server) / AJAX(browser to server) call support on SSR frontend to AD platform
  * No need to use swagger
  * Technical supports are great

**Comment by sfc-gh-nalekseev** [comment]
![SNO-SnowflakeLogo_blue](https://user-images.githubusercontent.com/108667117/218988787-f77f4a88-b3cc-4fdc-9fa3-031bbe1edbeb.png)


- Company Name: Snowflake
- Product/service name that adopted Armeria: 10+ internal services
- Any websites: www.snowflake.com
- The initiative adopting Armeria:
    - gRPC support
    - Ability to easily plug into existing applications
    - Highly customizable with decorators
- Top 3 benefits in the use case
    - Easy to configure and customize
    - Integrates well with other libraries for monitoring, logging
    - Great support from the maintainers

**Comment by prototype01** [comment]
- Company Name
  - Companion System

- logo
  ![image](https://user-images.githubusercontent.com/16009250/235821584-be7f2233-979a-49af-91da-7ccc2a1601b1.png)

- Product/service name that adopted Armeria
  - Petaso

- Any websites
  -  http://www.cpst.co.kr/

 - The initiative adopting Armeria
   - Tiny web framework
   - Easy for C#, golang developers to learn
   - Highly customizable http request

 - Top 3 benefits in the use case
   - Team members adapt quickly (It's easier than the spring)
   - gRPC and http api have the same port
   - Correspond to a variety of situations (like a financial site, They use http protocol but not standard)

- Our additional story
  - We are that specializes in image processing and distribute processing engineering. use language C#, C++, Rust, golang
  - But Most customers wanted a JVM environment. Container environment also did not support it
  - So we had to learn spring, but spring was too high an abstraction 
  - Armeria was the solution to us. It was easy to learn and easy to customize because it was a tiny and powerful framework

**Comment by ohadgur** [comment]
![OB Logo](https://github.com/line/armeria/assets/11676929/bcdc6be9-8f5b-498e-9d8b-d64d1dfdfb17)
- Company Name:
  -  Outbrain Inc. 

- Product/service name that adopted Armeria: 
   - Our microservices' RPC communication is using Armeria's WebClient as its underlying http client.
   - Using the http client wrapper as the default client for outgoing traffic.
   
- Websites:
  -   https://www.outbrain.com/ (Official Site)
  -   https://medium.com/outbrain-engineering (Medium)

- The initiative adopting Armeria: 
  -  Non-blocking, async, Http2-supporting client
  -  Flexible API
  -  Actively maintained library

- Top 3 benefits in the use case:
  - Flexibility and ease of use of the API
  - Excellent and fast support/feedback from the maintainers on slack/github
  - We were able to move away from using an old, EOL library

**Comment by neuru-lee** [comment]
<img width="706" alt="image" src="https://github.com/line/armeria/assets/108500058/68734503-b902-4325-859c-5511e5055610">

Company Name: Bucketplace

Product/service name that adopted Armeria:
- 10+ gRPC microservice backends that support 오늘의집 and Ohouse services.

Any websites: 
- [오늘의집](https://ohou.se/)
- [Ohouse - Home Styling Ideas](https://apps.apple.com/id/app/ohouse-home-styling-ideas/id1618890305)

The initiative adopting Armeria:
- Integration with Spring WebFlux
- gRPC support, including HTTP/JSON transcoding
- Support for Kotlin and Coroutines

Top 3 benefits in the use case:
- Possible to adopt Armeria at our own pace.
- Easy to build and customize a gRPC service.
- Easily share the API specification by providing a DocService URL that includes an example request.

**Comment by sh-cho** [comment]
Here's our use case :)

<img alt="kakao logo" src="https://github.com/line/armeria/assets/11611397/6043ee39-d868-4965-b058-118041038a19" width="400">

- Company name(+logo): kakao
- Product/service name that adopted Armeria
  - <img height="24" width="24" src="https://cdn.simpleicons.org/KakaoTalk" /> Several microservices of KakaoTalk messaging system
- Any websites: https://www.kakaocorp.com/
- The initiative adopting Armeria
  - Legacy RPC/custom-protocol app server migration
  - Support Kotlin coroutine
- Top 3 benefits in the use case
  - Easy to learn and have informative documentation
  - Easy to integrate RPC/HTTP/etc services
  - Enthusiastic contributors

**Comment by chrisryan10** [comment]
![RGB_CashApp_Logo_Primary_White](https://github.com/line/armeria/assets/73275196/f9ad1d0c-efc9-40fc-b739-fc4faaad9d7b)

-  Product/service name that adopted Armeria
   - Various microservices in the CashApp eco system
-  The initiative adopting Armeria
   - Providing more choices for our developers for various styles of development
-  Top 3 benefits in the use case
   - Good documentation making it easier to learn and adapt, plus great responses in Slack, and now Discord.
   - Good Kotlin support
   - Excellent GRPC/HTTP/etc support

**Reply by trustin** [reply]
Tweeted about it: https://x.com/armeria_project/status/1732579147313279122

**Reply by clamothebb** [reply]
@chrisryan10 I'm curious - Do you use this with Wire, or protobuf-java? I'd love to use Armeria with Wire, but ran into an issue when I tried.

**Reply by chrisryan10** [reply]
I've left the company now and retired, so I'm not sure what is the current state.  We did have Wire in use, although personally I did not believe Wire was worthwhile overall.

**Comment by leehow1988** [comment]
- Company name(+logo):  swiftpass 
  ![image](https://github.com/line/armeria/assets/6386894/2102f6c4-7ccf-49de-b239-59c3166f4c78)
- Product/service name that adopted Armeria
    -  Several microservices 
- Any websites: https://www.swiftpass.cn/
- The initiative adopting Armeria
    - Legacy RPC/custom-protocol app server migration
    - Ability to easily plug into existing applications
    - Highly customizable with decorators
    - Actively maintained library
- Top 3 benefits in the use case
    - Easy to learn and have informative documentation
    - Easy to integrate RPC/HTTP/etc services
    - Enthusiastic contributors

**Comment by novoj** [comment]
## Company name: [FG Forrest, a.s.](https://www.fg.cz)

![FG Forrest, a.s.](https://github.com/user-attachments/assets/55ad3cf8-dfa7-495b-9b09-9829594b2ffe)

## Product/service name: [evitaDB](https://evitadb.io)

![EvitaDB](https://github.com/user-attachments/assets/17f7e1ac-d004-4dd8-9b79-a82c62d5d570)

**The initiative adopting Armeria:**

- unify all supported web APIs (gRPC, REST, GraphQL, plain/old servlets) under a single web server.
- open the gRPC protocol for use by browsers (gRPC/web)
- actively maintained library

**Top 3 benefits in the use case:**

- run all APIs on a single port
- run HTTP and HTTPs on the same port
- get automatic documentation and debugger for gRPC protocol
- personal support and consultations via the Discord server

**Comment by evolved37** [comment]
# Gullivernet × Armeria Use Case 🙌

Here is our use case!

[![Gullivernet Logo](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTcMZkRA7kRa3BHEgsUbnfeMtqbyLBNobxYrA&s)](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTcMZkRA7kRa3BHEgsUbnfeMtqbyLBNobxYrA&s)

---

## Company Information

- **Company Name:** Gullivernet  
- **Website:** [https://www.gullivernet.com/](https://www.gullivernet.com/)  
- **Product/Service using Armeria:** Server for REST API and socket services of the HR platform (G-HR)

---

## G-HR Product Overview

**G-HR** is an integrated solution for managing people and workplaces, designed for both **SMBs** and **large enterprises**.  
It covers daily activities such as:

- attendance  
- shifts  
- vacation/leave requests  
- expense reports  

The platform provides **real-time monitoring** and reporting to optimize administrative processes.  
📊 **Results:** 40 hours saved per month · 200+ client companies · 100k users

---

## Key Features

- **Planning and timesheet:** manage projects/activities and increase productivity.  
- **Space booking:** reserve meeting rooms, desks, and company resources with real-time visualization.  
- **Access control and time tracking:** including off-site, with integration to **G-Badge** for virtual badges.  
- **Emergency and safety management:** people counting and safety protocols.  
- **Internal communication:** news, surveys, file manager, and reports.

---

## Integrations and AI

- **Single Sign-On** and services with **Microsoft** and **Google**  
- **Auth0**, **Mapspeople** (map-based bookings)  
- Integration with **Copilot AI**  
- Welfare module with **Valyouness**  
- G-HR can work as a **stand-alone solution** or integrate with other Gullivernet products.

---

## Initiative Adopting Armeria 🚀

- Reliable **HTTP/2 protocol** support  
- Event loop–based **non-blocking server**

---

## Top 3 Benefits in This Use Case 🌟

1. Build a **client factory** with decorators, protocol, and discovery support  
2. Easy use of the HTTP implementation by wrapping the HTTP reactive interface  
3. Improved performance and scalability for API and socket services

---

✍️ _Gullivernet Team_
